core/crypto/vrf/vrf_provider_impl.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/vrf/vrf_provider_impl.hpp"
+
+namespace kagome::crypto {
+
+  VRFKeypair VRFProviderImpl::generateKeypair() const {
+    size_t initseed = std::random_device()();
+    std::mt19937 gen(initseed);
```

Comment:
Never do this.

Use libp2p/crypto/random_generator

Inject it and use here.

Reply:
Why not use kagome::crypto::random::CSPRNG, more precisely kagome::crypto::random::BoostGeneratorImpl
It uses safe and fast cryptographic generator. It has `randomBytes(size_t)` function which returns vector of generated random bytes.